subsys/bluetooth/mesh/access.c

```diff
@@ -2140,16 +2166,23 @@ static int read_comp_cb(const char *key, size_t len, settings_read_cb read_cb,
 	return -EALREADY;
 }
 
-int bt_mesh_comp_read(struct net_buf_simple *buf)
+int bt_mesh_comp_read(struct net_buf_simple *buf, uint8_t page)
 {
+	if (page > 1) {
+		return -EINVAL;
+	}
+
+	char subtree[13];
 	size_t original_len = buf->len;
 	int err;
 
 	if (!IS_ENABLED(CONFIG_BT_SETTINGS)) {
 		return -ENOTSUP;
 	}
 
-	err = settings_load_subtree_direct("bt/mesh/cmp", read_comp_cb, buf);
+	sprintf(subtree, "bt/mesh/cmp%d", page);
```

Comment:
Do you really need `sprintf` and `stdio` for this? How about concatenation?
`printf` family of functions is quite heavy.

Reply:
I will look into this 👍